pkg/k8shandler/statefulset.go

```diff
@@ -641,6 +644,12 @@ func (node *statefulSetNode) isChanged() bool {
 			changed = true
 		}
 
+		if !comparators.EnvValueEqual(desiredContainer.Env, nodeContainer.Env) {
```

Comment:
Same observation as in deployment.go about the "// Only Image and Resources..." comment

Reply:
Yes, that will also trigger a rolling restart.